### PR TITLE
✨ Add v2 Rewards and Achievements + icons

### PR DESCRIPTION
**Pull request by matt-goldman**
In this PR:

* Some general clean-up
* Adds icons for achievements
* Adds seeder for icon data
* Adds prefix for rewards and achievements (to reduce round-trips when scanning in the mobile app)
* Adds logic to award milestone achievements (badges)

### CODE REVIEW

File: API/SSW.Rewards.Application/Users/Commands/UploadProfilePic/UploadProfilePicCommand.cs [hunk 1]
```diff
@@ -3,51 +3,81 @@
...
+                    .ThenInclude(ua => ua.Achievement)
+                .AsNoTracking()
+                .FirstOrDefaultAsync(cancellationToken);
+            
+            user.Avatar = url;
+
+            var achievement = await _context.Achievements.FirstOrDefaultAsync(a => a.Name == MilestoneAchievements.ProfilePic, cancellationToken);
+
+            if (!user.UserAchievements.Any(a => a.Achievement == achievement))
+            {
```
Comment (lukecookssw) [inline API/SSW.Rewards.Application/Users/Commands/UploadProfilePic/UploadProfilePicCommand.cs:67]: You don't need to go get the achievement from the db here.
Just change the check to be:
if (!user.UserAchievements.Any(a => a.Achievement.Name == MilestoneAchievements.ProfilePic))

File: API/SSW.Rewards.Application/Users/Commands/UploadProfilePic/UploadProfilePicCommand.cs [hunk 2]
```diff
@@ -3,51 +3,81 @@
...
+                user.UserAchievements.Add(new UserAchievement
+                {
+                    Achievement = achievement
+                });
+
```
Comment (lukecookssw) [inline API/SSW.Rewards.Application/Users/Commands/UploadProfilePic/UploadProfilePicCommand.cs:72]: When you get the user on line 55, you're doing it AsNoTracking(). Then you're modifying the user by adding this achievement. Isn't that wrong?

File: API/SSW.Rewards.Application/Users/Commands/UploadProfilePic/UploadProfilePicCommand.cs [hunk 3]
```diff
@@ -3,51 +3,81 @@
...
+            string url = await _storage.UploadProfilePic(bytes, filename);
+            
+            var user = await _context.Users
+                .Where(u => u.Email.ToLower() == _currentUserService.GetUserEmail())
+                .Include(u => u.UserAchievements)
+                    .ThenInclude(ua => ua.Achievement)
+                .AsNoTracking()
+                .FirstOrDefaultAsync(cancellationToken);
+            
+            user.Avatar = url;
```
Comment (lukecookssw) [inline API/SSW.Rewards.Application/Users/Commands/UploadProfilePic/UploadProfilePicCommand.cs:62]: AsNoTracking() then setting the Avatar. EF should be tracking the user in this case, no?

File: Xamarin/SSW.Rewards/SSW.Rewards/Services/RewardService.cs [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+namespace SSW.Rewards.Services
+{
+    public class RewardService : BaseService, IRewardService
+    {
+        private RewardClient _rewardClient;
+
+        public RewardService()
+        {
+            _rewardClient = new RewardClient(BaseUrl, AuthenticatedClient);
+        }
```
Comment (lukecookssw) [inline Xamarin/SSW.Rewards/SSW.Rewards/Services/RewardService.cs:15]: Where is the IHttpClientFactory for the RewardClient?

**Review by lukecookssw — COMMENTED** [review]
see comments

File: API/SSW.Rewards.Application/Users/Commands/UploadProfilePic/UploadProfilePicCommand.cs (hunk 2 above)
Comment (matt-goldman) [reply API/SSW.Rewards.Application/Users/Commands/UploadProfilePic/UploadProfilePicCommand.cs:72]: Done

File: API/SSW.Rewards.Application/Users/Commands/UploadProfilePic/UploadProfilePicCommand.cs (hunk 1 above)
Comment (matt-goldman) [reply API/SSW.Rewards.Application/Users/Commands/UploadProfilePic/UploadProfilePicCommand.cs:67]: Done.  Moved the call to get the achievement inside the if block.

File: API/SSW.Rewards.Application/Users/Commands/UploadProfilePic/UploadProfilePicCommand.cs (hunk 3 above)
Comment (matt-goldman) [reply API/SSW.Rewards.Application/Users/Commands/UploadProfilePic/UploadProfilePicCommand.cs:62]: Done.

**Review by lukecookssw — APPROVED** [review]
lgtm

**Pull request merged**